Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs

```diff
@@ -1715,10 +1715,15 @@ private void ChangePhaseOfActionInternal(int actionIndex, TriggerState* actionSt
 
             // Update action state.
             var newState = trigger;
+
+            // We need to update the magnitude before we copy the flags across as the copied flags may contain
+            // the `HaveMagnitude` flag which would give us an incorrect magnitude value. (case 1239551)
```

Comment:
Now I notice this is actually incorrect this way, too. It computes the magnitude which sets the `HaveMagnitude` flag on `newState`. Which may then get nuked by the copying over of flags.

I'll go fix it and merge the PR.